naga/src/proc/keyword_set.rs

```diff
@@ -0,0 +1,192 @@
+use core::{fmt, hash};
+
+use crate::racy_lock::RacyLock;
+use crate::FastHashSet;
+
+/// A case-sensitive set of strings,
+/// for use with [`Namer`][crate::proc::Namer] to avoid collisions with keywords and other reserved
+/// identifiers.
+///
+/// This is currently implemented as a hash table.
+/// Future versions of Naga may change the implementation based on speed and code size
+/// considerations.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct KeywordSet(FastHashSet<&'static str>);
+
+impl KeywordSet {
+    /// Returns a new mutable empty set.
+    #[inline]
```

Comment:
Now with small functions being default inline, I don't think we need to mark each one as inline.

Reply:
Whoops. I must have been adding these on autopilot.

Even after that compiler change, I’ve been burned several times by "this basic function is not inlined in a hot loop and fixing that gets me -20% run time". But, these functions outside of `contains()` very definitely should *not* be in any hot loop, and many of them are generic (therefore always at least inlin*able*) anyway.

Removed all of the `#[inline]`s except on `contains()`.